src/query/util/execution/parallel_test.go

```diff
@@ -78,13 +79,34 @@ func TestOrderedParallel(t *testing.T) {
 }
 
 func TestSingleError(t *testing.T) {
+	defer leaktest.Check(t)()
```

Comment:
nit: Is this necessary? I think CI runs with `--race` but I may be mistaken as to what this does

Reply:
Ah yeah, this is serving a different purpose--it checks for goroutines that leak after the end of the test. For instance (from their README):

// Default "Check" will poll for 5 seconds to check that all
// goroutines are cleaned up
func TestPool(t *testing.T) {
    defer leaktest.Check(t)()

    go func() {
        for {
            time.Sleep(time.Second)
        }
    }()
}